apollo-sample/src/main/java/com/example/apollographql/sample/MainActivity.java

```diff
@@ -18,25 +21,31 @@
 public class MainActivity extends AppCompatActivity {
   private static final String TAG = "MainActivity";
 
+  private SampleApplication application;
+
   @Override protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_main);
     final TextView txtResponse = (TextView) findViewById(R.id.txt_response);
-    SampleApplication application = (SampleApplication) getApplication();
-    application.githuntApiService()
-        .githuntFeed(new FeedQuery(FeedQuery.Variables.builder()
-                .limit(10)
-                .type(FeedType.HOT)
-                .build()))
+    application = (SampleApplication) getApplication();
+
+    Observable<Response<FeedQuery.Data>> query = Observable.fromCallable(new Callable<Response<FeedQuery.Data>>() {
+      @Override public Response<FeedQuery.Data> call() throws Exception {
+        return executeFeedQuery();
+      }
+    });
+
+    query
         .subscribeOn(Schedulers.io())
         .observeOn(AndroidSchedulers.mainThread())
         .subscribe(new Observer<Response<FeedQuery.Data>>() {
```

Comment:
please use a `DisposableObserver` which does not have the `OnSubscribe` callback